pkg/catalogv2/git/utils.go

```diff
@@ -0,0 +1,148 @@
+package git
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	stateDir  = "management-state/git-repo"
+	staticDir = "/var/lib/rancher-data/local-catalogs/v2"
+	localDir  = "../rancher-data/local-catalogs/v2" // identical to helm.InternalCatalog
+)
+
+func gitDir(namespace, name, gitURL string) string {
+	staticDir := filepath.Join(staticDir, namespace, name, hash(gitURL))
+	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
+		return staticDir
+	}
+	localDir := filepath.Join(localDir, namespace, name, hash(gitURL))
+	if s, err := os.Stat(localDir); err == nil && s.IsDir() {
+		return localDir
+	}
+	return filepath.Join(stateDir, namespace, name, hash(gitURL))
+}
+
+func isBundled(git *git) bool {
+	return strings.HasPrefix(git.Directory, staticDir) || strings.HasPrefix(git.Directory, localDir)
+}
+
+// isGitSSH checks if the URL is in the SSH URL format using regular expressions.
+// [anything]@[anything]:[anything]
+// ssh://<user>@<mydomain.example>:<port>/<path>/<repository-name>
+func isGitSSH(gitURL string) bool {
+	pattern1 := `^[^:/]+@[^:]+:[a-zA-Z]+/[^/]+$`
+	pattern2 := `^ssh://[^@]+@[^:]+:\d+/.+$`
+
+	// Check if the input matches either of the two patterns.
+	valid, err := regexp.MatchString(pattern1, gitURL)
+	if err != nil {
+		return false
+	}
+	if valid {
+		return true
+	}
+	valid, err = regexp.MatchString(pattern2, gitURL)
+	if err != nil {
+		return false
+	}
+
+	return valid
+}
+
+// validateGitURL will validate if the provided URL is in one of the expected patterns
+// for the supported protocols http(s) or ssh.
+//   - if Valid: returns nil
+//   - if Invalid: returns an error
+func validateGitURL(gitURL string) error {
+	valid := isGitSSH(gitURL)
+	if valid {
+		return nil
+	}
+	// not ssh; validate http(s)
+	u, err := url.Parse(gitURL)
+	if err != nil || (u.Scheme != "http" && u.Scheme != "https") {
+		return fmt.Errorf("invalid git URL: %s", gitURL)
```

Comment:
The error message should say invalid URL right?

Reply:
OK

Reply:
@rohitsakala please resolve this